File: 2022/1/src/LargestSumCalories.py
# declare list to hold sum of calories held by each elf
# declare variable to hold each line of file
# declare variable to hold the largest sum of calories 
# the file name is "input.txt"
# try to open the file, if it fails, print an error message and exit
# read each line of the file and add it to a temporary variable
# when you encounter a new line, store the value of the temporary variable in the list as a new element
# repeat until the end of the file
# once the file has been read, close the file
# find the largest sum of calories in the vector
# print the largest sum of calories

# Path: src/LargestSumCalories.py
caloriesSum = []
line = ""
largestSum = 0
# the file name is "input.txt"
try:
    f = open("input.txt", "r")
except:
    print("Error: could not open file")
    exit()
# read each line of the file and add it to a temporary variable
for x in f:
    line += x
    # when you encounter a new line, store the value of the temporary variable in the list as a new element
    if x == "\n": 
        caloriesSum.append(int(line))
        line = ""
# find the largest sum of calories in the vector
for x in caloriesSum:
    if x > largestSum:
        largestSum = x
# print the largest sum of calories
print(largestSum)

File: 2022/1/src/LargestSumCalories.cpp
/* LargestSumCalories.cpp
  Author: Kian Dadkhah
  Date: 2023-03-13
  Description: This program reads a file containing a list of calories held by each 'elf" in a group. The file is formatted such that each line contains the calories held by one elf, and each group of elves is separated by a blank line. The program then calculates the sum of calories held by each group of elves, and prints the maximum sum.
  
  Example input file:
  100
  200
  300

  400
  500
  600

  700

  800
  900
  1000
  1100


  Example output:
  Max sum: 2100
  

  Note: This program assumes that the input file is formatted correctly. It does not check for errors in the input file.
*/

#include <iostream>
#include <fstream>
#include <vector>
#include <string>

using namespace std;

int main() {
  // declare vector to hold sums
  vector<int> sumCalories;
  // declare variable to hold max sum
  int maxSum = 0;

  // open file
  ifstream inFile("input.txt");
  // check if file opened successfully
  if (!inFile) {
    cout << "Unable to open file";
    exit(1); // terminate with error
  }

  // read input file line by line
  string line;
  int sum = 0;
  while (getline(inFile, line)) {
    // if line is empty, add sum to vector and reset sum
    if (line.empty()) {
      sumCalories.push_back(sum);
      sum = 0;
    } else {
      // add calories to sum
      sum += stoi(line);
    }
  }
  // add last sum to vector
  sumCalories.push_back(sum);

  // close file
  inFile.close();

  // find max sum
  for (int i = 0; i < sumCalories.size(); i++) {
    if (sumCalories[i] > maxSum) {
      maxSum = sumCalories[i];
    }
  }

  // print max sum
  cout << "Max sum: " << maxSum << endl;

  return 0;
}